design/notifications.md

```diff
@@ -0,0 +1,360 @@
+      description: The reference type. Either BRANCH or TAG.
+    referenceName:
+      type: string
+      description: The deleted reference name.
+```
+
+#### Example
+```json
+{
+  "type": "REFERENCE_DELETED",
+  "eventTime": "2009-02-14T18:09:10Z",
+  "referenceType": "BRANCH",
+  "referenceName": "main"
+}
+```
+
+## Webhooks/Notifications API
+A webhook is a method to augment the behavior of a web service with custom callbacks. These callbacks are usually a URL
+that a client registers with the webservice, in order for it to get invoked by the webservice whenever there's a
+relevant event.
+
+In Nessie's case, the contract between the webservice (caller) and the webhook URL (callee) is a simple HTTP request
+sent to the registered webhook. The contract is as follows:
+1. The client will register a webhook notification with Nessie by calling one of the APIs below.
+2. When an event happens in Nessie, it will call back using a `POST` request that:
+   1. Its content-type is `application/json`.
+   2. Its body contains the JSON payload of the event that happened. (See previous section for details on the expected schema).
+
+### Creating a new notification/webhook
+There are different event types a client might want to get notified about. In order to subscribe to a particular event 
+type, a client needs to make a `POST` request to one or more of the following endpoints.
+
+* `/notifications/commits`
+* `/notifications/merges`
+* `/notifications/transplants`
+* `/notifications/references-created`
+* `/notifications/references-deleted`
+* `/notifications/references-assigned`
+
+The request body is expected to have the following schema
+```yaml
+Notification:
+  type: object
+  properties:
+    type:
+      type: string
+      description: The type of notification. Currently only "WEBHOOK" is supported.
+    url:
+      type: string
+      format: uri
+      description: The url to callback.
+```
+
+A webhook notification creation request might look like
+
+```json
+{
+  "type": "WEBHOOK",
+  "url": "http://www.example.org/webhook-callback"
+}
+```
+
+Nessie will respond with a `201 Created` response and a `Location` header corresponding to the URI of the recently
+created notification. This resource URI will be used for deleting, updating and describing the notification.
+
+```
+HTTP/1.1 201 Created
+Location: http://nessie.dremio.site/api/v1/notifications/6bbdcc62-e5b6-4e1b-8f06-3b0bb3620d5e
+Content-type: application/json
+
+{
+  "id": "6bbdcc62-e5b6-4e1b-8f06-3b0bb3620d5e",
+  "type": "WEBHOOK",
+  "url": "http://www.example.org/webhook-callback"
+}
+```
+
+Important note is that nessie doesn't make any efforts to de-duplicate the URL and if a particular URL is subscribed
+twice to the same type of event. Nessie will notify twice, once per each notification.
+
+### Describing, updating and deleting notifications
+As previously described, when creating a notification, the response will contain the URI of the newly created 
```

Comment:
how do we get a list of notifications that we own?

Reply:
I will add a `GET /notifications`